include/picongpu/simulation/stage/RuntimeDensityFile.hpp

```diff
@@ -0,0 +1,124 @@
+/* Copyright 2022 Sergei Bastrakov
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+#include "picongpu/particles/Particles.hpp"
+#include "picongpu/particles/boundary/Kind.hpp"
+#include "picongpu/particles/densityProfiles/FromOpenPMDImpl.hpp"
+
+#include <pmacc/Environment.hpp>
+#include <pmacc/meta/ForEach.hpp>
+#include <pmacc/particles/traits/FilterByFlag.hpp>
+
+#include <boost/program_options.hpp>
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+
+namespace picongpu
+{
+    namespace simulation
+    {
+        namespace stage
+        {
+            namespace detail
+            {
+                /** Functor to set runtime density file for the given species via command-line parameters
+                 *
+                 * @tparam T_Species particle species type
+                 */
+                template<typename T_Species>
+                class RuntimeDensityFileCommandLine
+                {
+                public:
+                    //! Create runtime density file functor
+                    RuntimeDensityFileCommandLine() : prefix(T_Species::FrameType::getName())
+                    {
+                    }
+
+                    /** Register command-line options
+                     *
+                     * Done as operator() to simplify invoking with pmacc::meta::ForEach
+                     */
+                    void operator()(po::options_description& desc)
+                    {
+                        // Density from openPMD is conditionally enabled, so use same condition
+#if(ENABLE_OPENPMD == 1)
+                        auto* filename = &densityProfiles::RuntimeDensityFile<T_Species>::get();
+                        desc.add_options()(
+                            (prefix + "_runtimeDensityFile").c_str(),
```

Comment:
Is this option always visible even if the user is not using the density from OpenPMD functor?

I am asking because if this option is always visible it could be the thinks it is always possible to use this command line parameter and then wondering why no density is created.

Reply:
It is always visible. There is explanation in help and docs. I also made the option name longer e.g. not `--e_densityFile`

Reply:
I generally agree with the concern, but didn't see a reasonable way to do it otherwise. As usual, coupling between our compile- and runtime stuff is awkward